### PR TITLE
Python: build and link to our libbz2 under Linux and Windows

### PR DESCRIPTION
**Pull request by dbrnz**
Note that the CMake that runs as part of `ExternalProject_Add` is not seeing the `C_COMPILER` and `CXX_COMPILER` environment variables (because it runs at build time, not configure time??) so:
```
-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
```
was needed.

### CODE REVIEW

**Comment by agarny** [comment]
@dbrnz: it looks like there is a problem with the Python package on Linux...?

**Comment by dbrnz** [comment]
This is because a new pre-built Python package is required, but only for Linux. I could upload one and update the appropriate SHAs but then the regular Linux build would fail...

**Comment by agarny** [comment]
I am happy for a new pre-built Python package to be uploaded, so that this PR can pass. Ok, I am going to do that, no problem.

File: src/plugins/thirdParty/Python/CMakeLists.txt [hunk 1]
```diff
@@ -79,6 +79,12 @@ endif()
 
 set(EXTERNAL_BINARIES_DIR ${FULL_LOCAL_EXTERNAL_PACKAGE_DIR}/${LIBRARY_DIR})
 
+# Make sure the directory exists
+
+file(MAKE_DIRECTORY ${EXTERNAL_BINARIES_DIR})
```
Comment (agarny) [inline src/plugins/thirdParty/Python/CMakeLists.txt]: Why do we need to do this? I mean, we don't do need anywhere else, so...?

**Comment by dbrnz** [comment]
Windows will also need new pre-built packages, incorporating the statically linked `libbz2.lib`.

**Comment by agarny** [comment]
> Windows will also need new pre-built packages, incorporating the statically linked `libbz2.lib`.

Ok, will look into it too.

File: src/plugins/thirdParty/Python/CMakeLists.txt (hunk 1 above)
Comment (dbrnz) [reply src/plugins/thirdParty/Python/CMakeLists.txt]: Hmm, historically it was happening as part of setting up `sqlite3` so I moved where it was done. But you are right, it's not needed...

**Comment by agarny** [comment]
Hmm... trying to build the Python package on Windows, I am getting the following error:

```
ninja: error: FindFirstFileExA(../ext/src/plugins/thirdparty/python/release/libs): The directory name is invalid.
```

and, indeed, I can see that there is a (binary) file (not a directory) called `Libs`...

**Comment by dbrnz** [comment]
I'm now wondering if there was a reason for:
```
file(MAKE_DIRECTORY ${EXTERNAL_BINARIES_DIR})
```

**Pull request merged**